subworkflows/local/scatter_genome/main.nf

```diff
@@ -0,0 +1,68 @@
+include { BEDTOOLS_MERGE           } from '../../../modules/nf-core/bedtools/merge/main'
+include { BEDTOOLS_SORT            } from '../../../modules/nf-core/bedtools/sort/main'
+include { BUILD_INTERVALS          } from '../../../modules/local/build_intervals/main'
+include { SPLIT_BED_CHUNKS         } from '../../../modules/local/split_bed_chunks/main'
+
+workflow SCATTER_GENOME {
+
+    take:
+    ch_fai             // channel: [optional] [ val(meta), path(fai) ]
+    ch_input_bed       // channel: [optional] [ val(meta), path(bed) ]
+    make_bed_from_fai  // bool
+    make_bed_intervals // bool
+    split_n            // integer: split bed into n regions
+
+    main:
+    ch_versions = Channel.empty()
+    ch_bed = Channel.empty()
+    ch_bed_intervals = Channel.empty()
+
+    //
+    // If no BED-file is provided then build intervals from reference
+    //
+    if( make_bed_from_fai ) {
+
+        ch_fai
+            .set{ ch_build_intervals_in }
+
+        BUILD_INTERVALS ( ch_build_intervals_in )
+        ch_versions = ch_versions.mix(BUILD_INTERVALS.out.versions)
+
+        BUILD_INTERVALS.out.bed
+            .set{ ch_bed }
+    } else {
+        ch_input_bed
+            .set{ ch_bed }
+    }
+
+    //
+    // Merge overlapping and then split BED regions for SNV calling
+    //
+    if( make_bed_intervals ) {
+
+        if( split_n < 1 ) { error "Can't split bed file into less than one file" }
```

Comment:
I noticed you have set a minimum value for this paramater in nextflow.config. Doesn't nf-validation throw an error if you provide a value below one?

Reply:
It does yes. 

I started adding some nf-test to the subworkflows, having the check inside here would allow the subworkflow to stand on it's own a bit more...But I can remove this check and the test that is supposed to fail, because it does become a double check in the pipeline and I'm not sure what's appropriate.

Reply:
Tbh, I don't have strong opinion on this, but I don't think its inappropriate with having a double check. I was more curious to know if nf-test was equipped to handle such checks on it own or not. 😅

Reply:
Alright, then I'll keep it like this for now although it's a bit inconsistent, and we can clear it up at a later time depending on the state of nf-test/core.